### ISSUE TITLE
UI doesn't render in Ableton Live 10

### ISSUE THREAD

**Issue by alalalexexex** [antonok-edm/ampli-Fe#5]
Steps for repro:
 - Built using `cargo build --release`
 - copied .dll file to main folder for Ableton plugins
 - Open plugin as normal, but the screen is blank (white canvas).
 
![not_working](https://user-images.githubusercontent.com/43417713/131230649-cf2a3ded-e71e-48c3-8a83-77be881c211f.PNG)

*Note: I'm a complete novice for vst dev, and just wanted to explore where to start: i.e. In no rush, but wanted to see if I could compile something from scratch.

**Comment by antonok-edm** [comment]
Thanks for the report! I don't have access to Ableton myself so this is good to know.

This is probably another issue with [`vst_window`](https://crates.io/crates/vst_window); I suppose Ableton does something different from the other DAWs I've tested.

**Comment by antonok-edm** [comment]
@alalalexexex or others, if you have a Mac it'd be interesting to see if `ampli-Fe` is also broken with Ableton there.

**Comment by skyelights** [comment]
I'm using Ableton live 11 and running a release build of ampli-Fe makes it crash (Windows)

**Comment by skyelights** [comment]
@antonok-edm I finally had a minute to test it on Ableton 11 on a Mac and it seems to work without any issues, so strange that it doesn't seem to function as well on the Windows version.

<img width="646" alt="image" src="https://user-images.githubusercontent.com/752642/132327085-e621a970-f80b-4f7f-a19a-0e71bfc26634.png">

**Comment by detrimentalist** [comment]
Same issue with Bitwig 4.0.5 on Ubuntu 20.04

![Screenshot_2022-01-17_11-32-40](https://user-images.githubusercontent.com/65347538/149753667-9bf2d66a-820b-455d-ac69-95badb261618.png)

**Comment by antonok-edm** [comment]
@detrimentalist after loading an instance of ampli-Fe, could you check your Bitwig's engine logs for any errors? Should be in `~/.BitwigStudio/log/engine.log` by default.

I'm using Bitwig 4.1.0 in Arch without any issues, so I suspect what you're seeing is a hardware or driver related issue in `wgpu`.

**Comment by antonok-edm** [comment]
@detrimentalist yep, that's roughly what I expected to see. If one of the adapters works, but isn't automatically chosen by `instance.request_adapter(...)`, then it's an issue with `wgpu` or your installed driver/hardware combination. You might be running into something like https://github.com/gfx-rs/wgpu/issues/2397 - it'd be worth commenting there or opening another report.

Granted, ampli-Fe is pulling a version of `wgpu` from more than a year ago, so it may also be worth trying to update that.

**Comment by antonok-edm** [comment]
@detrimentalist could you try building the [`update-wgpu` branch](https://github.com/antonok-edm/ampli-Fe/tree/update-wgpu)? That's using the latest published `wgpu` version.

(I suppose others who were facing rendering issues are encouraged to try as well :smile:)

**Comment by detrimentalist** [comment]
Building from the update-wgpu branch resolved the issue for me. Thanks!

**Comment by faejr** [comment]
I finally got around to building the update-wgpu branch and I'm unfortunately still having issues. Although I did have to also make the following patches to get it to build: https://github.com/antonok-edm/vst_window/pull/2

![image](https://user-images.githubusercontent.com/752642/151674145-d4859059-c78d-430f-9df1-14b9680563b3.png)
